test/unit/lib/plugins/aws/package/compile/events/apiGateway/lib/hack/updateStage.test.js

```diff
@@ -809,4 +809,66 @@ describe('test/unit/lib/plugins/aws/package/compile/events/apiGateway/lib/hack/u
     expect(untagResourceStub).to.have.been.calledOnce;
     expect(untagResourceStub.args[0][0].tagKeys).to.deep.equal(['keytoremove']);
   });
+
+  it('should deploys shouldStartNameWithService without apiName', async () => {
```

Comment:
I only have one minor comment - in general, it's better to not override service for tests, as it's automatically generated - I see that here we need service for assertion so we can go this route instead:

```
const { serviceConfig, servicePath: serviceDir, updateConfig } = await fixtures.setup('apiGateway');
// Here we can get service from serviceConfig
await updateConfig(); // Here we can pass config ext content
await runServerless({cwd: serviceDir, ...}) // change fixture to cwd 
```

Reply:
Should I also replace my `const service = "...";` with `serviceConfig.service`?

Reply:
yes, I don't remember exactly if under `service` there will be name or an object so you might need to check if `serviceConfig.service.name` won't be needed there

Reply:
It's a plain string, I've update the test.